src/main/java/views/add_edit_category_views/EditCategoryV.java

```diff
@@ -87,8 +87,9 @@ public void actionPerformed(ActionEvent evt) {
         //Two ActionListeners with different behaviours differentiated by checking evt.getSource().
         if (evt.getSource() == categoryCombo) {
             this.selectedCategory = (String) categoryCombo.getSelectedItem();
-        }
-        else {
+        } else if(categoryCombo.getSelectedIndex() == -1){
+            JOptionPane.showMessageDialog( this, "Please select a category you wish to edit.");
```

Comment:
Maybe you could try if it works without having this else if clause, because the same code didn't work in your EditExpenseV.

Reply:
I've abandoned this if statement to address the error, please take a look in my next commit.

Reply:
I see that it's been removed from EditExpenseV, but unfortunate it also doesn't work in here. When I tried to edit a category without clicking the dropdown list, it would just add a new month rather than warning the user to select a category. I think you could try replacing 
line 23:  
private String selectedCategory;             with            private String selectedCategory = "";
and
line 90: 
else if(categoryCombo.getSelectedIndex() == -1)           with          if(selectedCategory == "")

Reply:
That should be a problem in EditCategory. I realized I didn't have that one updated. I'm not sure what you meant by "it added you a new month", may you elaborate?

Reply:
**Yes, its a problem in EditCategory.** (Sorry it was a bit confusing. I said EditExpenseV you previously had similar code in EditExpenseV too.)

From my understanding, with this code you are trying to enforce the user to select a Category, and thereby separating this EditCategory case from AddCategory case. However, **there is a problem with this code, and it does not notify the user that they haven't selected a Category**. As a result, your controller handles this "have not selected a Category to edit" case as AddCategory case (because **the selectedCategory inputted is null**), and created a new Category instead. 

Hopefully the attached screenshots can give you an idea of what I'm talking about. 
<img width="528" alt="Screenshot 2023-08-13 171345" src="https://github.com/CSC207-2023Y-UofT/course-project-DoorKeepers/assets/133046046/7c5c4e41-9841-4af3-8923-1562cb002e33">
<img width="520" alt="Screenshot 2023-08-13 171400" src="https://github.com/CSC207-2023Y-UofT/course-project-DoorKeepers/assets/133046046/5b599672-210b-4402-80ec-86e4875a6568">

Reply:
I think it's actually solved in your latest commit, so I'll mark this as resolved.